src/onegov/org/views/ticket.py

```diff
@@ -1069,6 +1074,72 @@ def action_link(ticket):
     }
 
 
+@OrgApp.html(model=ArchivedTicketsCollection, name='delete',
+             request_method='DELETE', permission=Secret)
+def view_delete_all_archived_tickets(self, request):
+    tickets = self.query().filter_by(state='archived')
+
+    errors, ok = delete_tickets_and_related_data(request, tickets)
+    if errors:
+        msg = request.translate(_(
+            "${success_count} tickets deleted, "
+            "${error_count} are not deletable",
+            mapping={'success_count': len(ok), 'error_count': len(errors)},
+        ))
+        request.message(msg, 'warning')
+    else:
+        msg = request.translate(_(
+            "${success_count} tickets deleted.",
+            mapping={'success_count': len(ok)}
+        ))
+        request.message(msg, 'success')
+
+
+def delete_tickets_and_related_data(
+    request: 'CoreRequest', tickets: 'Query[Ticket]'
+) -> tuple[list['Ticket'], list['Ticket']]:
+
+    not_deletable, successfully_deleted = [], []
+
+    for ticket in tickets:
+        if (hasattr(
+                ticket.handler, 'ticket_deletable')
+                and not ticket.handler.ticket_deletable):
```

Comment:
```suggestion
        if not getattr(ticket.handler, 'ticket_deletable', True):
```
`hasattr` + attribute access can be simplified to `getattr` with default.

Although I think the default should be flipped, it doesn't really seem safe to me to delete a ticket when the handler doesn't specify whether it should be deleted or not.

Reply:
But also I would honestly just add `ticket_deletable` with a sane default to the base handler class, then you don't need `getattr`/`hasattr` at all.